paddlex/inference/pipelines_new/image_multilabel_classification/pipeline.py

```diff
@@ -0,0 +1,88 @@
+from ...common.reader import ReadImage
+from ...common.batch_sampler import ImageBatchSampler
+from ...utils.pp_option import PaddlePredictorOption
+from ..base import BasePipeline
+
+# [TODO] 待更新models_new到models
+from ...models_new.image_multilabel_classification.result import MLClassResult
+
+
+class ImageMultiLabelClassificationPipeline(BasePipeline):
+    """Image Multi Label Classification Pipeline"""
+
+    entities = "image_multilabel_classification"
+
+    def __init__(
+        self,
+        config: Dict,
+        device: str = None,
+        pp_option: PaddlePredictorOption = None,
+        use_hpip: bool = False,
+        hpi_params: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        """
+        Initializes the class with given configurations and options.
+
+        Args:
+            config (Dict): Configuration dictionary containing model and other parameters.
+            device (str): The device to run the prediction on. Default is None.
+            pp_option (PaddlePredictorOption): Options for PaddlePaddle predictor. Default is None.
+            use_hpip (bool): Whether to use high-performance inference (hpip) for prediction. Defaults to False.
+            hpi_params (Optional[Dict[str, Any]]): HPIP specific parameters. Default is None.
+        """
+        super().__init__(
+            device=device, pp_option=pp_option, use_hpip=use_hpip, hpi_params=hpi_params
+        )
+
+        self.threshold = config["SubModules"]["ImageMultiLabelClassification"].get(
+            "threshold", None
+        )
+        image_multilabel_classification_model_config = config["SubModules"][
+            "ImageMultiLabelClassification"
+        ]
+        self.image_multilabel_classification_model = self.create_model(
+            image_multilabel_classification_model_config
+        )
+        batch_size = image_multilabel_classification_model_config["batch_size"]
+        self.batch_sampler = ImageBatchSampler(batch_size=batch_size)
+        self.img_reader = ReadImage(format="BGR")
+
+    def predict(
+        self,
+        input: str | list[str] | np.ndarray | list[np.ndarray],
+        threshold: float | dict | list | None = None,
+        **kwargs
+    ) -> MLClassResult:
+        """Predicts image classification results for the given input.
+
+        Args:
+            input (str | list[str] | np.ndarray | list[np.ndarray]): The input image(s) or path(s) to the images.
+            **kwargs: Additional keyword arguments that can be passed to the function.
+
+        Returns:
+            TopkResult: The predicted top k results.
+        """
+
+        for img_id, batch_data in enumerate(self.batch_sampler(input)):
+            batch_imgs = self.img_reader(batch_data)
+            for mlclas_single_result in self.image_multilabel_classification_model(
+                input=batch_imgs,
+                threshold=self.threshold if threshold is None else threshold,
+            ):
+                yield mlclas_single_result
```

Comment:
yield from self.image_multilabel_classification_model(input=batch_imgs, threshold=self.threshold if threshold is None else threshold)